### PR TITLE
gpu job template

### PR DESCRIPTION
**Pull request by asafyehezkel**
- Add tollerations to engine gpu jobs
- Add volumes and volumes mount "or conditions"

### CODE REVIEW

**Review by yotamazriel — APPROVED** [review]
LGTM

Let's change the toleration interface in the next PRs

File: charts/engine/templates/engine-job-template-gpu-cm.yaml [hunk 1]
```diff
@@ -37,6 +37,12 @@ data:
 {{ if .Values.schedulerName }}
           schedulerName: {{ .Values.schedulerName }}
 {{ end }}
+{{ if .Values.tolerations }}
```
Comment (yotamazriel) [inline charts/engine/templates/engine-job-template-gpu-cm.yaml:42]: Let's just rename it to add gpu tolerations

Comment (asafyehezkel) [reply charts/engine/templates/engine-job-template-gpu-cm.yaml]: done

File: charts/engine/templates/engine-job-template-gpu-cm.yaml [hunk 2]
```diff
@@ -24,7 +24,9 @@ data:
               envFrom:
                 - configMapRef:
                     name: engine-cm
+{{- if or .Values.additional_pvcs .Values.localDataDirectory }}      
               volumeMounts:
+{{- end}}
```
Comment (arikmaor) [inline charts/engine/templates/engine-job-template-gpu-cm.yaml:29]: Please also add this change to the other engine job template (the `cpu` one)

File: charts/engine/templates/engine-job-template-gpu-cm.yaml [hunk 3]
```diff
@@ -37,7 +39,15 @@ data:
 {{ if .Values.schedulerName }}
           schedulerName: {{ .Values.schedulerName }}
 {{ end }}
+{{ if .Values.addGpuTolerations }}
+          tolerations:
+            - key: nvidia.com/gpu
+              operator: Exists
+              effect: NoSchedule
+{{ end}}
```
Comment (arikmaor) [inline charts/engine/templates/engine-job-template-gpu-cm.yaml:47]: I don't think we should add this

**Review by arikmaor — COMMENTED** [review]
10x!

Few comments inside:
1. empty `volumes` - nice, please add also to the `cpu` engine template
2. `toleration` - seems unnecessary, but if we do add them, we should allow the user to specify them
3. *DON'T FORGET* to update `engine` and `tensorleap` chart versions before merging

File: charts/engine/templates/engine-job-template-gpu-cm.yaml (hunk 2 above)
Comment (asafyehezkel) [reply charts/engine/templates/engine-job-template-gpu-cm.yaml:29]: Done, I did it to `volumes` as well unless do you think we shouldn’t

File: charts/engine/templates/engine-job-template-gpu-cm.yaml (hunk 3 above)
Comment (asafyehezkel) [reply charts/engine/templates/engine-job-template-gpu-cm.yaml:47]: Might be, @yotamazriel and I search what are the changes between running jobs manifests(alongside `run-ai` agent) with not running jobs manifests.
The toleration was one of the differences

**Pull request closed without merging**